docs/package_templates/header_only/all/conanfile.py

```diff
@@ -50,30 +50,31 @@ def layout(self):
 
     def requirements(self):
         # prefer self.requires method instead of requires attribute
-        self.requires("dependency/0.8.1")
+        # direct dependencies of header only libs are always transitive since they are included in public headers
+        self.requires("dependency/0.8.1", transitive_headers=True)
```

Comment:
@memsharded @lasote Could you tell us whether transitive_libs should also be True here? I think so but I'm not sure.

Reply:
Not true. You can consume any project and only include its headers on your .cpp or internal headers, don't need to expose it by default.

Reply:
A header-only lib has no cpp or private headers by definition.

Reply:
Fair enough.

Reply:
@SpaceIm in case you want to add more comments: https://github.com/conan-io/conan/issues/12364